includes/framework/QDbBackedSessionHandler.class.php

```diff
@@ -123,6 +123,15 @@ public static function SessionRead($id) {
 			if (!$result_row) // either the data was empty or the row was not found
 				return '';
 			$strData = $result_row['data'];
+
+			if(strstr($objDatabase->Adapter, 'PostgreSql')) {
```

Comment:
IMHO, no. That code is run everywhere! For it to work correctly in FetchArray method in Postgres adapter, the information about the datatypes of each field in the row is to be passed to the method as well - which is too big a task considering the complexity of the rest of the framework.